http/codegen/server.go

```diff
@@ -1324,7 +1324,7 @@ const responseT = `{{ define "response" -}}
 	{{- end }}
 
 	{{- if .ErrorHeader }}
-	w.Header().Set("goa-error", {{ printf "%q" .ErrorHeader }})
+	w.Header().Set("goa-error", res.ErrorName())
```

Comment:
I had to make this change so that we always use this `res` variable that we initialize in the error encoder. There could be a situation where the `res` variable is initialized and never used causing compile issues.